grandeur/download_files_ica.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+'''
+Author: Erin Young
+
+Description:
+
+This script will download the results from ica
+
+EXAMPLE:
+python3 download_files_ica.py -o <directory to download to> -r <run name>
+'''
+
+# trying to keep dependencies really low
+import argparse
+import subprocess
+import concurrent.futures
+import os
+
+version = '0.0.20230925'
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-o', '--out', type=str, help='directory where files will get downloaded to', default='ica')
+parser.add_argument('-r', '--run', type=str, help='sequencing run', required = True )
+parser.add_argument('-v', '--version', help='print version and exit', action='version', version='%(prog)s ' + version)
+args   = parser.parse_args()
+
+
+# getting id 
+# icav2 projectdata list --file-name UT-M03999-230830 --match-mode FUZZY --parent-folder /
+ica_lines = subprocess.check_output(['icav2', 'projectdata', 'list', '--file-name', args.run + '_Gran', '--match-mode', 'FUZZY', '--parent-folder', '/'], universal_newlines= True, text='str')
+for line in ica_lines.split('\n'):
+    if 'AVAILABLE' in line:
+        folder_id=line.split("\t")[3]
+
```

Comment:
This way of grabbing the ICA id should work for most runs unless there were multiple runs. In which case the first folders will be the newest runs. This may not be an issue because you will probably only be downloading results if you know the run completed successfully but I did want to point this out.